src/connectors/NetworkConnector.ts

```diff
@@ -22,19 +22,83 @@ class RequestError extends Error {
   }
 }
 
+interface BatchItem {
+  request: { jsonrpc: '2.0'; id: number; method: string; params: unknown }
+  resolve: (result: any) => void
+  reject: (error: Error) => void
+}
+
 class MiniRpcProvider implements AsyncSendable {
```

Comment:
batches eth_estimateGas, eth_gas and potentially eth_blockNumber though that is usually sequential with the other two